test_common/KokkosKernels_TestUtils.hpp

```diff
@@ -288,41 +356,27 @@ void vanillaGEMM(typename ViewTypeC::non_const_value_type alpha,
   }
 }
 
-template <class ViewTypeA, class ViewTypeX, class ViewTypeY>
-void vanillaGEMV(char mode, typename ViewTypeA::non_const_value_type alpha,
-                 const ViewTypeA& A, const ViewTypeX& x,
-                 typename ViewTypeY::non_const_value_type beta,
-                 const ViewTypeY& y) {
+template <class AlphaType, class ViewTypeA, class ViewTypeX, class BetaType,
+          class ViewTypeY>
+KOKKOS_INLINE_FUNCTION void vanillaGEMV(char mode, AlphaType alpha,
+                                        const ViewTypeA& A, const ViewTypeX& x,
+                                        BetaType beta, const ViewTypeY& y) {
   using ScalarY = typename ViewTypeY::non_const_value_type;
   using KAT_A   = Kokkos::ArithTraits<typename ViewTypeA::non_const_value_type>;
-  using KAT_Y   = Kokkos::ArithTraits<ScalarY>;
-  int M         = A.extent(0);
-  int N         = A.extent(1);
-  if (beta == KAT_Y::zero()) Kokkos::deep_copy(y, KAT_Y::zero());
-  if (mode == 'N') {
-    for (int i = 0; i < M; i++) {
-      ScalarY y_i = beta * y(i);
-      for (int j = 0; j < N; j++) {
-        y_i += alpha * A(i, j) * x(j);
-      }
-      y(i) = y_i;
-    }
-  } else if (mode == 'T') {
-    for (int j = 0; j < N; j++) {
-      ScalarY y_j = beta * y(j);
-      for (int i = 0; i < M; i++) {
-        y_j += alpha * A(i, j) * x(i);
-      }
-      y(j) = y_j;
-    }
-  } else if (mode == 'C') {
+  const bool transposed = mode == 'T' || mode == 'C';
+  const bool conjugated = mode == 'C';
+  const bool has_beta   = beta != Kokkos::ArithTraits<BetaType>::zero();
+  int M                 = A.extent(transposed ? 1 : 0);
```

Comment:
You should call `A.extent_int` here or you will probably get some casting warning with picky compilers.

Reply:
Thank you, fixed on https://github.com/kokkos/kokkos-kernels/pull/1433/commits/ad631ae28a56b6e83198176a6aaa9d85ea28fbd9.